src/modules/advancedOrders/updaters/RecipientUpdater.tsx

```diff
@@ -0,0 +1,20 @@
+import { useUpdateAtom } from 'jotai/utils'
+import { useEffect } from 'react'
+
+import useENSAddress from 'legacy/hooks/useENSAddress'
+
+import { useAdvancedOrdersDerivedState } from '../hooks/useAdvancedOrdersDerivedState'
+import { updateAdvancedOrdersAtom } from '../state/advancedOrdersAtom'
+
+export function RecipientUpdater() {
```

Comment:
Great!

Reply:
Should we do the same for all trade widgets?

Reply:
Addressed this in https://github.com/cowprotocol/cowswap/pull/2886